src/allmydata/web/status.py

```diff
@@ -35,187 +35,236 @@ def render_time(self, ctx, data):
     def render_rate(self, ctx, data):
         return abbreviate_rate(data)
 
-class UploadResultsRendererMixin(RateAndTimeMixin):
+
+class UploadResultsRendererMixin(Element):
     # this requires a method named 'upload_results'
 
-    def render_pushed_shares(self, ctx, data):
+    @renderer
+    def pushed_shares(self, req, tag):
         d = self.upload_results()
-        d.addCallback(lambda res: res.get_pushed_shares())
+        d.addCallback(lambda res: str(res.get_pushed_shares()))
         return d
```

Comment:
Might be better to use `u"{}".format(res.get_pushed_shares())` instead? (So that it's unicode on py2 and py3?)

Reply:
Leaving this as it is for now, considering the follow-up comment below.